File: Kattis/8queens.py
def main():
    a, b = [False]*15, [False]*15
    c, d = [False]*8, [False]*8

    q = 0
    for i in range(8):
        for j in range(8):
            pos = input()
            while (pos != "*" and pos != "."):
                pos = input()

            if pos == "*":
                q += 1

                if a[j-i+7] or b[i+j] or c[i] or d[j]:
                    return "invalid"
                    
                a[j-i+7] = True
                b[i+j] = True
                c[i] = True
                d[j] = True
    
    if q == 8:
        return "valid"
    else:
        return "invalid"

if __name__ == "__main__":
    print(main())

File: Kattis/8queens.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(0); cin.tie(0); cout.tie(0);

    vector<bool> a(15, false);
    vector<bool> b(15, false);
    vector<bool> c(15, false);
    vector<bool> d(15, false);

    int q = 0;
    char pos;

    for (int i = 0; i < 8; ++i) {
        for (int j = 0; j < 8; ++j) {
            cin >> pos;
            if (pos == '*') {
                ++q;
                
                if (a[j-i+7] || b[i+j] || c[i] || d[j]) {
                    cout << "invalid";
                    return 0; 
                }

                a[j-i+7] = true;
                b[i+j] = true;
                c[i] = true;
                d[j] = true;
            }
        }
    }
    if (q == 8) cout << "valid";
    else cout << "invalid";

    return 0;
}